src/ui/groups/common/GroupForm.js

```diff
@@ -25,122 +23,157 @@ const msgs = defineMessages({
   },
 });
 
-export class GroupFormBody extends Component {
-  componentWillMount() {
-    this.props.onWillMount(this.props);
-  }
+export const GroupFormBody = (props) => {
+  const {
+    intl, isValid, isSubmitting: submitting, mode, submitForm,
+    dirty, onCancel, onDiscard, onHideModal, values, onDidMount,
+    setFieldValue, setFieldTouched, onWillUnmount,
+  } = props;
+  useEffect(() => {
+    onDidMount(props);
+    return () => onWillUnmount();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-  onSubmit = (ev) => {
-    ev.preventDefault();
-    this.props.handleSubmit();
-  };
+  const isEdit = mode === EDIT_MODE;
 
-  render() {
-    const {
-      intl, invalid, submitting, mode, onChangeName, onFocus,
-      dirty, onCancel, onDiscard, onSave,
-    } = this.props;
+  useEffect(() => {
+    if (!isEdit) {
+      setFieldValue('code', values.name.replace(/\W/g, '_').toLowerCase());
+      setFieldTouched('code', true);
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
is it really necessary here?

Reply:
lint requires to add the used dispatch props in useEffect array parameter but I don't think it's necessary since our function does not change during runtime